lib/configuration.ex

```diff
@@ -1,61 +1,80 @@
 defmodule Configuration do
+  alias Game.Board, as: Board
+  alias Player.Human, as: Human
+  alias Player.Computer, as: Computer
 
-  def configure_game do
-    {player_1, player_2} = set_players()
-    board_size = get_board_size()
+  def configure_game(configuration \\ Configuration, console \\ Console) do
+    {player_1, player_2} = configuration.get_players(console)
+    board_size = configuration.get_board_size(console)
     %Game{board: Board.create(board_size), 
           size: board_size,
           player_1: player_1,
           player_2: player_2,
           current_player: player_1}
   end
 
-  def set_players(console \\ Console) do
-    game_type_input = console.get_game_type
-    if game_type_input |> Validator.is_valid_option? do
-      game_type_input 
-      |> Integer.parse 
-      |> elem(0) 
-      |> get_marks()
-    else
-      %Message{}.invalid |> console.print
-      set_players()
+  def get_board_size(console \\ Console) do
+    board_size_input = console.get_board_size
+    board_size_input 
+    |> Validator.is_valid_option?
+    |> set_board_size(board_size_input, console)
+  end
+
+  def set_board_size(valid?, board_size_input, console \\ Console, on_invalid_input \\ &get_board_size/1)
+
+  def set_board_size(true, board_size_input, _console, _on_invalid_input) do
+    case board_size_input do
+      "1" -> 3
+      "2" -> 4
     end
   end
+
+  def set_board_size(false, _board_size_input, console, on_invalid_input) do
+    %Message{}.invalid |> console.print
+    on_invalid_input.(console)
+  end
+
+  def get_players(console) do
+    game_type_input = console.get_game_type
+    game_type_input
+    |> Validator.is_valid_option? 
```

Comment:
This might be a nitpick, but I'm curious how it would clean up your code if you made your Validator a guard instead of returning true/false.

If false -> try again
if true -> return the input and move on

This way, your code would look a little cleaner
```
console.get_game_type
|> Validator.check_is_valid_option
|> set_players
```